### PR TITLE
Config init+

### PR DESCRIPTION
**Pull request by bcoca**
Giving ansible-config a few new features

 - can now list/dump for specific plugins (not just dump the whole list of builtins)
 - new 'init' option that can create a highly commented example config file, env vars or yaml vars file

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

ansible-config
##### ADDITIONAL INFORMATION
```shell
#> ansible-config init  
```
```ini
[defaults]
# By default Ansible will issue a warning when received from a task action (module or action plugin)
# These warnings can be silenced by adjusting this setting to False.
action_warnings=True

# White list of cowsay templates that are 'safe' to use, set to empty list if you want to enable all installed templates.
cowsay_enabled_stencils=bud-frogs, bunny, cheese, daemon, default, dragon, elephant-in-snake, elephant, eyes, hellokitty, kitty, luke-koala, meow, milk, moofasa, moose, ren, sheep, small, stegosaurus, stimpy, supermilker, three-eyes, turkey, turtle, tux, udder, vader-koala, vader, www

# Specify a custom cowsay path or swap in your cowsay implementation of choice
cowpath=None

# This allows you to chose a specific cowsay stencil for the banners or use 'random' to cycle through them.
cow_selection=default

# This option forces color mode even when running without a TTY or the "nocolor" setting is True.
force_color=False

# This setting allows suppressing colorizing output, which is used to give a better indication of failure and status information.
nocolor=False

# If you have cowsay installed but want to avoid the 'cows' (why????), use this.
nocows=False

# Sets the default value for the any_errors_fatal keyword, if True, Task failures will be considered fatal errors.
any_errors_fatal=False

# Colon separated paths in which Ansible will search for Become Plugins.
become_plugins=~/.ansible/plugins/become:/usr/share/ansible/plugins/become

# Chooses which cache plugin to use, the default 'memory' is ephemeral.
fact_caching=memory

# Defines connection or path information for the cache plugin
fact_caching_connection=None

# Prefix to use for cache plugin files/tables
fact_caching_prefix=ansible_facts

# Expiration timeout for the cache plugin data
fact_caching_timeout=86400

...

[netconf_connection]
# This variable is used to enable bastion/jump host with netconf connection. If set to True the bastion/jump host ssh settings should be present in ~/.ssh/config file, alternatively it can be set to custom ssh configuration file path to read the bastion/jump host settings.
ssh_config=None


[paramiko_connection]
# TODO: write it
host_key_auto_add=False

# TODO: write it
look_for_keys=True


[jinja2]
# This list of filters avoids 'type conversion' when templating variables
# Useful when you want to avoid conversion into lists or dictionaries for JSON strings, for example.
dont_type_filters=string, to_json, to_nice_json, to_yaml, to_nice_yaml, ppretty, json

```

### CODE REVIEW

**Comment by briantist** [comment]
Have not looked over this in detail or tried to use it locally, but I like the idea, as the maintainer of a collection making heavy use of plugin ini/env/vars for its options; thanks for this!

File: lib/ansible/cli/config.py [hunk 1]
```diff
@@ -176,11 +195,10 @@ def _list_plugin_settings(self, ptype):
 
         return entries
 
-    def execute_list(self):
+    def _list_entries_from_args(self):
         '''
-        list all current configs reading lib/constants.py and shows env and config file setting names
+        build a dic with the list requested configs
```
Comment (felixfontein) [inline lib/ansible/cli/config.py]:
```suggestion
        build a dict with the list requested configs
```

File: changelogs/fragments/config_updates.yml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+minor_changes:
+  - ansible-config can now list and dump for specific documentable plugins by specifying them in the command line
+  - ansible-config has new 'init' option to create, highly commented, example configurations as ini, env vars or yaml variable files
```
Comment (felixfontein) [inline changelogs/fragments/config_updates.yml]: The comments in the code say that YAML isn't supported yet.

File: lib/ansible/plugins/loader.py [hunk 3]
```diff
@@ -914,10 +914,16 @@ def all(self, *args, **kwargs):
             basename = os.path.basename(name)
 
             if basename == '__init__' or basename in _PLUGIN_FILTERS[self.package]:
+                # either empty or contains base classes
```
Comment (felixfontein) [inline lib/ansible/plugins/loader.py]:
```suggestion
                # either empty or ignored by the module blocklist
```
BTW, the blocklist is still called blacklist in https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/loader.py#L1064=L1086

File: changelogs/fragments/config_updates.yml (hunk 2 above)
Comment (bcoca) [reply changelogs/fragments/config_updates.yml]: yaml is 'vars in yaml file' .. not the yaml key .. but will reword

**Comment by samdoran** [comment]
!waffling core_review

**Review by Shrews — APPROVED** [review]

**Pull request merged**